cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -40,13 +41,13 @@ case class CommandLineBindingCommandPart(argument: CommandLineBinding) extends C
                            runtimeEnvironment: RuntimeEnvironment): ErrorOr[InstantiatedCommand] =
     Try {
       val pc = ParameterContext(runtime = runtimeEnvironment.cwlMap).withInputs(inputsMap.map({
-        case (LocalName(localName), WomSingleFile(path)) => localName -> WomString(path)
+        case (LocalName(localName), sf: WomSingleFile) => localName -> valueMapper(sf)
         case (LocalName(localName), value) => localName -> value
       }), functions)
 
       val womValue: WomValue = argument match {
         case CommandLineBinding(_, _, _, _, _, Some(StringOrExpression.Expression(expression)), Some(false)) =>
-          expression.fold(EvaluateExpression).apply(pc)
+          expression.fold(EvaluateExpression).apply(pc) |> valueMapper
```

Comment:
some more things needed value mapping

Reply:
Sigh... I thought the whole point of `cats` was that they did away with all these nonsense operators

Reply:
Personally I have learned to love the right turn arrow's effect on readability. 😄

Reply:
After a short stay in Room `1|>1`?